config/mapprovider.go

```diff
@@ -81,20 +81,39 @@ type ChangeEvent struct {
 
 // Retrieved holds the result of a call to the Retrieve method of a Provider object.
 type Retrieved struct {
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.MergeTo to access.
 	*Map
-
-	// CloseFunc specifies a function to be invoked when the configuration for which it was
-	// used to retrieve values is no longer in use and should close and release any watchers
-	// that it may have created.
-	//
-	// If nil, then nothing to be closed.
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.Close to access.
 	CloseFunc
 }
 
-// CloseFunc a function to close and release any watchers that it may have created.
+// NewRetrievedFromMap returns a new Retrieved instance that contains a Map data.
+// * cfgMap the Map that will be merged to the given map in the MergeTo.
+// * CloseFunc specifies a function to be invoked when the configuration for which it was
+//   used to retrieve values is no longer in use and should close and release any watchers
+//	 that it may have created.
+func NewRetrievedFromMap(cfgMap *Map, close CloseFunc) Retrieved {
+	return Retrieved{Map: cfgMap, CloseFunc: close}
+}
+
+// MergeTo merges the retrieved configuration into the given Map.
+// Note that the retrieved configuration must not be modified.
+func (r Retrieved) MergeTo(dest *Map) error {
```

Comment:
Should we handle the case where `dest` is nil explicitly and return an error?

Reply:
> We don't do that in lots of other places... 

That's fair, I was thinking about the close function case, but it seems like we mostly only do this for function pointers.

> Let's have an issue about validating the input arguments, and make a decision for the entire repo?

Sounds good, I am now doubtful about what the best approach is but it seems like a good idea to have the discussion in any case.